### PR TITLE
test: Receiving inbox message tests

### PR DESCRIPTION
**Pull request by RasmusGraabaek**
## Description

Adding a test to verify that we are able to receive a message via the inbox que.

Next step: will be to processing AggregatedMeassureDataRequest with mock data in order to connect EDI and Wholesale

## References

https://app.zenhub.com/workspaces/mosaic-60a6105157304f00119be86e/issues/gh/energinet-datahub/team-phoenix/60

### CODE REVIEW

File: .github/workflows/ci-dotnet.yml [hunk 1]
```diff
@@ -85,6 +85,12 @@ jobs:
       tests_dll_file_path: \source\dotnet\wholesale-api\Events\Events.IntegrationTests\bin\Release\net7.0\Energinet.DataHub.Wholesale.Events.IntegrationTests.dll
       download_attempt_limit: 20
       use_azure_functions_tools: true
+      environment: AzureAuth
+      run_integration_tests: true
```
Comment (BjarkeMeier) [inline .github/workflows/ci-dotnet.yml:89]: Just curious. How come this is needed here but not present in e.g. step dotnet_ci_test_integrationtests_batches?

File: source/dotnet/wholesale-api/Events/Events.Application/Workers/AggregatedTimeSeriesServiceBusWorker.cs [hunk 2]
```diff
@@ -27,16 +29,18 @@ public class AggregatedTimeSeriesServiceBusWorker : BackgroundService, IAsyncDis
...
     private ServiceBusProcessor? _serviceBusProcessor;
 
     public AggregatedTimeSeriesServiceBusWorker(
         IAggregatedTimeSeriesRequestHandler aggregatedTimeSeriesRequestHandler,
         ILogger<AggregatedTimeSeriesRequestHandler> logger,
-        string serviceBusConnectionString)
+        IOptions<ServiceBusOptions> options)
     {
-        _serviceBusClient = new ServiceBusClient(serviceBusConnectionString);
+        _serviceBusClient = new ServiceBusClient(options.Value.SERVICE_BUS_LISTEN_CONNECTION_STRING);
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/Events/Events.Application/Workers/AggregatedTimeSeriesServiceBusWorker.cs:40]: What are your findings about instantiating multiple service bus clients (there is at least one other instance in the host)?

File: source/dotnet/wholesale-api/Events/Events.Application/Workers/AggregatedTimeSeriesServiceBusWorker.cs [hunk 3]
```diff
@@ -57,7 +61,7 @@ public async ValueTask DisposeAsync()
 
     protected override async Task ExecuteAsync(CancellationToken stoppingToken)
     {
-        _serviceBusProcessor = _serviceBusClient.CreateProcessor("sbq-wholesale-inbox");
+        _serviceBusProcessor = _serviceBusClient.CreateProcessor(_options.INBOX_MESSAGE_QUEUE_NAME);
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/Events/Events.Application/Workers/AggregatedTimeSeriesServiceBusWorker.cs:64]: Missing a `using`?

File: source/dotnet/wholesale-api/Events/Events.IntegrationTests/Fixture/ServiceBusSenderFixture.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    async Task IAsyncLifetime.DisposeAsync()
+    {
+        await DisposeAsync();
+        await Task.CompletedTask.ConfigureAwait(false);
+    }
+
+    internal Task PublishAsync(string message)
+    {
+        var client = new ServiceBusClient(ServiceBusOptions.Value.SERVICE_BUS_LISTEN_CONNECTION_STRING);
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/Events/Events.IntegrationTests/Fixture/ServiceBusSenderFixture.cs:71]: `using`?

File: source/dotnet/wholesale-api/Events/Events.IntegrationTests/Fixture/ServiceBusSenderFixture.cs [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        var sender = client.CreateSender(ServiceBusOptions.Value.INBOX_MESSAGE_QUEUE_NAME);
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/Events/Events.IntegrationTests/Fixture/ServiceBusSenderFixture.cs:72]: `using`?

File: source/dotnet/wholesale-api/Events/Events.Application/Workers/AggregatedTimeSeriesServiceBusWorker.cs (hunk 3 above)
Comment (MadsDue) [reply source/dotnet/wholesale-api/Events/Events.Application/Workers/AggregatedTimeSeriesServiceBusWorker.cs:64]: AggregatedTimeSeriesServiceBusWorker is impl IAsyncDisposable and is also responsible for call dispose on _serviceBusProcessor

File: source/dotnet/wholesale-api/Events/Events.IntegrationTests/Fixture/ServiceBusSenderFixture.cs [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        return sender.SendMessageAsync(CreateAggregatedTimeSeriesRequestMessage(message));
+    }
+
+    private ServiceBusMessage CreateAggregatedTimeSeriesRequestMessage(string body)
+    {
+        var message = new ServiceBusMessage(body);
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/Events/Events.IntegrationTests/Fixture/ServiceBusSenderFixture.cs:78]: I anticipate that meta data will be added in another PR.

File: source/dotnet/wholesale-api/Events/Events.IntegrationTests/Infrastructure/AggregatedTimeSeriesRequests/AggregatedTimeSeriesRequestsTests.cs [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        _sender = fixture;
+    }
+
+    [Theory]
+    [InlineAutoMoqData]
+    public async Task Can_Receive_aggregated_time_series_request_service_bus_message(
+        Mock<IAggregatedTimeSeriesRequestHandler> handlerMock,
+        Mock<ILogger<AggregatedTimeSeriesRequestHandler>> loggerMock)
+    {
+        var sut = new AggregatedTimeSeriesServiceBusWorker(
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/Events/Events.IntegrationTests/Infrastructure/AggregatedTimeSeriesRequests/AggregatedTimeSeriesRequestsTests.cs:40]: In this domain we use the AAA pattern. So please add the comments like
`// Arrange`
etc.

File: source/dotnet/wholesale-api/Events/Events.IntegrationTests/Fixture/ServiceBusSenderFixture.cs (hunk 6 above)
Comment (MadsDue) [reply source/dotnet/wholesale-api/Events/Events.IntegrationTests/Fixture/ServiceBusSenderFixture.cs:78]: correct

File: source/dotnet/wholesale-api/Events/Events.Application/Workers/AggregatedTimeSeriesServiceBusWorker.cs (hunk 2 above)
Comment (MadsDue) [reply source/dotnet/wholesale-api/Events/Events.Application/Workers/AggregatedTimeSeriesServiceBusWorker.cs:40]: I have talk with Søren about this tomorrow at 8:30. If you are avaliable I will call you in as well.

File: .github/workflows/ci-dotnet.yml (hunk 1 above)
Comment (MadsDue) [reply .github/workflows/ci-dotnet.yml:89]: Azure CLI authentication fails if we don't have them.

File: source/dotnet/wholesale-api/Events/Events.Application/Workers/AggregatedTimeSeriesServiceBusWorker.cs (hunk 3 above)
Comment (BjarkeMeier) [reply source/dotnet/wholesale-api/Events/Events.Application/Workers/AggregatedTimeSeriesServiceBusWorker.cs:64]: But is there a reason to make the processer an instance variable and have to manage the disposal of it when it could be a local variable automatically disposed of by a `using` statement?

File: source/dotnet/wholesale-api/Events/Events.IntegrationTests/Infrastructure/AggregatedTimeSeriesRequests/AggregatedTimeSeriesRequestsTests.cs [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    private readonly ServiceBusSenderFixture _sender;
+
+    public AggregatedTimeSeriesRequestsTests(ServiceBusSenderFixture fixture)
+    {
+        _sender = fixture;
+    }
+
+    [Theory]
+    [InlineAutoMoqData]
+    public async Task Can_Receive_aggregated_time_series_request_service_bus_message(
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/Events/Events.IntegrationTests/Infrastructure/AggregatedTimeSeriesRequests/AggregatedTimeSeriesRequestsTests.cs:36]: Very nice improvement regarding the use of the `AutoResetEvent` 👍
I have some thoughts about a few minor improvements to this test. Nothing major, but you're welcome to call me if you're interested.

**Review by BjarkeMeier — APPROVED** [review]
Nice job guys

**Review by MadsDue — APPROVED** [review]

**Review by Renetnielsen — APPROVED** [review]
